app/uk/gov/hmrc/eusubsidycompliancefrontend/models/Undertaking.scala

```diff
@@ -56,4 +58,29 @@ case class Undertaking(
 
 object Undertaking {
   implicit val undertakingFormat: OFormat[Undertaking] = Json.format[Undertaking]
+  implicit val undertakingOptFormat: OFormat[Option[Undertaking]] = new OFormat[Option[Undertaking]] {
+    override def reads(json: JsValue): JsResult[Option[Undertaking]] =
```

Comment:
I've seen a bunch of formatter in `uk.gov.hmrc.eusubsidycompliancefrontend.models.json.digital` - the package object. 

Also think it's appropriate here too - just running it by in case you wanted to move it there.

Reply:
yeah the digital package has the formatter but they were all for Undertaking . This was only for Option[Undertaking] so i thought it should better sit here. I can move no prob for me

Reply:
Sounds grand 👌